File: AtlasWeb/Core/wwwroot/app/NebulaClient/StorageManager/InstanceManagement/TTObjectState.ts
import { Guid } from "../../Mscorlib/Guid";
import { TTObjectStateDef } from "../DefinitionManagement/TTObjectStateDef";

export class TTObjectState {
    public ObjectID: Guid;
    public StateDefID: Guid;
    public StateDef: TTObjectStateDef;
    public IsUndo: boolean;
    public BranchDateTimeTick: number;
    public BranchDate: Date;
    public PrevState: TTObjectState;
    public Description: string;
    public UserID: Guid;
    //public User: TTUser;
}

File: AtlasModel/Enterprise/TTObjectState.cs
using System;

namespace AtlasModel.Enterprise
{
    public partial class TTObjectState
    {
        public Guid StateDefId { get; set; }
        public Guid ObjectId { get; set; }
        public bool IsUndo { get; set; }
        public long BranchDateTimeTick { get; set; }
        public string Description { get; set; }
        public Guid? UserId { get; set; }
        public DateTime BranchDate { get; set; }
    }
}